tests/unit/io/test_dataset.py

```diff
@@ -55,3 +58,49 @@ def test_infer_list_dtype_unknown():
     df = pd.DataFrame({"col": [[], []]})
     dataset = Dataset(df, cpu=True)
     assert dataset.schema["col"].dtype.element_type.value == "unknown"
+
+
+@pytest.mark.parametrize("engine", ["csv", "parquet"])
+def test_dask_df_array_npy(tmpdir, datasets, engine):
+    paths = glob.glob(str(datasets[engine]) + "/*." + engine.split("-")[0])
+    # cannot have any null/NA entries
+    dataset = Dataset(Dataset(paths).to_ddf().compute().fillna(method="ffill"))
+    path = str(tmpdir / "result.npy")
+    dataset.to_npy(path)
+    nparr = np.load(path, allow_pickle=True)
+    numpy_arr = dataset.to_ddf().compute().to_numpy()
+    assert (nparr == numpy_arr).all()
+
+
+@pytest.mark.parametrize("append", [True, False])
+@pytest.mark.parametrize("engine", ["csv", "parquet"])
+def test_dask_df_array_npy_append(tmpdir, datasets, engine, append):
+    df = make_df(
+        {
+            "id": [1, 2, 3, 4, 5, 6],
+            "embed_1": [1, 2, 3, 4, 5, 6],
+            "embed_2": [1, 2, 3, 4, 5, 6],
+            "embed_3": [1, 2, 3, 4, 5, 6],
+        }
+    )
+    dataset = Dataset(df)
+    path = str(tmpdir / "result.npy")
+    dataset.to_npy(path, append=append)
+    nparr = np.load(path, allow_pickle=True)
```

Comment:
I will update though, because it is only required when you have an object so the last two test cases do not actually need that allow-pickle option. It was an artifact of copying and pasting the original test (then changing the data).